packages/augur-state/src/db/DB.ts

```diff
@@ -103,43 +120,51 @@ export class DB<TBigNumber> {
   /**
    * Syncs generic events and user-specific events with blockchain and updates MetaDB info.
    *
-   * @param {Augur<TBigNumber>} augur Augur object with which to sync
+   * TODO Make use of EventLogDBRouter
+   *
+   * @param {Provider} ethereumProvider Ethereum provider to use for syncing
+   * @param {Events} events Object used for getting and parsing logs from the blockchain
    * @param {number} chunkSize Number of blocks to retrieve at a time when syncing logs
-   * @param {number} blockstreamDelay Number of blocks by which blockstream is behind the blockchain
    */
-  public async sync(augur: Augur<TBigNumber>, chunkSize: number, blockstreamDelay: number): Promise<void> {
-    let dbSyncPromises = [];
-    const highestAvailableBlockNumber = await augur.provider.getBlockNumber();
-    for (let dbIndex in this.syncableDatabases) {
-      dbSyncPromises.push(
-        this.syncableDatabases[dbIndex].sync(
-          augur,
-          chunkSize,
-          blockstreamDelay,
-          highestAvailableBlockNumber
-        )
-      );
-    }
+  public async sync(ethereumProvider: Provider, events: Events, chunkSize: number): Promise<void> {
+    const startSyncBlockNumber = await this.getSyncStartingBlock();
+    const endSyncBlockNumber = await ethereumProvider.getBlockNumber() - this.blockstreamDelay;
+    let firstBlockNumberInChunk = startSyncBlockNumber;
 
-    for (let trackedUser of await this.trackedUsers.getUsers()) {
+    while (firstBlockNumberInChunk < endSyncBlockNumber) {
+      const lastBlockNumberInChunk = Math.min(firstBlockNumberInChunk + chunkSize, endSyncBlockNumber);
+      const rawLogsPromises:Array<Promise<Array<Log>>> = [];
+
+      // Get all generic event logs in chunk
+      console.log(`SYNCING all generic events from blocks ${firstBlockNumberInChunk} to ${lastBlockNumberInChunk}`);
+      rawLogsPromises.push(events.getLogs(this.genericEventNames, firstBlockNumberInChunk, lastBlockNumberInChunk));
+
+      // Get all user-specific event logs in chunk
       for (let userSpecificEvent of this.userSpecificEvents) {
-        let dbName = this.getDatabaseName(userSpecificEvent.name, trackedUser);
-        dbSyncPromises.push(
-          this.syncableDatabases[dbName].sync(
-            augur,
-            chunkSize,
-            blockstreamDelay,
-            highestAvailableBlockNumber
-          )
-        );
-      }
-    }
+        const additionalTopics: Array<string | Array<string>> = [];
+        additionalTopics.fill("", userSpecificEvent.numAdditionalTopics);
+        const trackedUsers = await this.trackedUsers.getUsers();
+        additionalTopics[userSpecificEvent.userTopicIndex] = trackedUsers.map(trackedUser => `0x000000000000000000000000${trackedUser.substr(2)}`);
```

Comment:
Feels like 0x00000000000000000000000 should be in a method or as a constant or something ... to prevent mistakes.

Reply:
Agreed, there are a few other places in the code where we're doing this, so it'd probably be worth turning this into a function. I'll add that.